nservicebus/upgrades/8to9/index.md

```diff
@@ -1,18 +1,103 @@
 summary: Instructions on how to upgrade NServiceBus from version 8 to version 9.
-reviewed: 2022-04-22
+reviewed: 2023-12-13
 component: Core
 isUpgradeGuide: true
 upgradeGuideCoreVersions:
  - 8
  - 9
 ---
 
-## SendOptions changes
+## Removed support for .NET Framework
 
-In NServiceBus version 8 and earlier, the indication that a message was marked for [immediate dispatch](/nservicebus/messaging/send-a-message.md#dispatching-a-message-immediately) was provided by `SendOptions.RequiredImmediateDispatch()`. In version 9, this method has been renamed to `SendOptions.IsImmediateDispatchSet()`.
+NServiceBus 9 no longer supports any version of the .NET Framework. Instead, it targets .NET 8 only (Read more about the [supported frameworks and platforms](/nservicebus/upgrades/supported-platforms.md)). Any component in NServiceBus 8 that is .NET Framework only (for example, the MSMQ transport)will not have a version that is compatible with NServiceBus 9. NServiceBus 8 will continue to be supported for use on the .NET Framework.
 
-## DataBus interface changes
+## SendOptions immediate dispatch changes
 
-Starting with NServiceBus version 8, it is mandatory to provide a serializer to the data bus configuration API. The `BinaryFormatterDataBusSerializer` has been removed and `SystemJsonDataBusSerializer` is the default option that's built-in.
+The method used to determine if [immediate dispatch](/nservicebus/messaging/send-a-message.md#dispatching-a-message-immediately) has been requested for a message has been renamed.
+
+snippet: core-8to9-immediate-dispatch
+
+## IManageUnitsOfWork has been removed
+
+The `IManageUnitsOfWork` API has been removed. Instead, a [pipeline behavior should be used to implement custom units of work](/nservicebus/pipeline/unit-of-work.md#implementing-custom-unit-of-work).
+
+snippet: core-8to9-uow
+
+## API to override machine name has changed
+
+The `RuntimeEnvironment.MachineNameAction` API [Override the machine name](/nservicebus/hosting/override-machine-name.md) has been removed. The replacement API is the `HostInfoSettings.UsingHostName` method.
+
+snippet: core-8to9-machinename
+
+## API to set additional audit metadata has changed
+
+The API to [set additional audit metadata](/nservicebus/operations/auditing.md#additional-audit-information) has been changed.
+
+snippet: core-8to9-audit-metadata
+
+## Dependency registration access in features renamed
+
+The property used to access container registrations in features has been renamed from `Container` to `Services`.
+
+snippet: core-8to9-di-features
+
+## Service collection extensions for backwards compatibility removed
+
+Service collection extensions to ease [the transition to Microsoft DI abstractions](/nservicebus/upgrades/7to8/dependency-injection.md#registercomponents-changes) have been removed. It is now required to use the registration APIs added in NServiceBus 8.
+
+snippet: core-8to9-di-shims
+
+## Making features depend on message driven subscriptions
```

Comment:
Do we want to document this in this way? We don't really want people to try and take that dependency at all. Otherwise, why did we make the class internal? Seems like this entry should be saying something more like "no longer possible to take this dependency" ?

Reply:
We decided to move it to the bottom